### PR TITLE
Darwin: A couple fixes

### PR DESCRIPTION
**Pull request by spwhitt**

### CODE REVIEW

File: pkgs/development/libraries/botan/generic.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@ stdenv.mkDerivation rec {
   buildInputs = [ python bzip2 zlib gmp openssl boost ];
 
   configurePhase = ''
-    python configure.py --prefix=$out --with-bzip2 --with-zlib ${if openssl != null then "--with-openssl" else ""} ${extraConfigureFlags}
+    python configure.py --prefix=$out --with-bzip2 --with-zlib ${if openssl != null then "--with-openssl" else ""} ${extraConfigureFlags} ${if stdenv.cc.isClang then "--cc=clang" else "" }
```
Comment (wmertens) [inline pkgs/development/libraries/botan/generic.nix]: this has the effect of inserting an extra space for linux and therefore a rebuild.

File: pkgs/tools/misc/bandwidth/default.nix [hunk 2]
```diff
@@ -19,7 +19,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ nasm ];
 
-  buildFlags = [ arch ];
+  buildFlags = [ arch ]
+    ++ stdenv.lib.optionals stdenv.cc.isClang [ "CC=clang" "LD=clang" ];
```
Comment (wmertens) [inline pkgs/tools/misc/bandwidth/default.nix]: This comes back a lot, I wonder if stdenv shouldn't be defining CC and LD instead... @copumpkin?

**Comment by wmertens** [comment]
LGTM except for that extra space...

**Comment by spwhitt** [comment]
Extra space has been removed.

File: pkgs/tools/misc/bandwidth/default.nix (hunk 2 above)
Comment (spwhitt) [inline pkgs/tools/misc/bandwidth/default.nix]: I think stdenv has no way of knowing that CC and LD are valid make flags - they're a convention, not a rule.

The reason this comes up so often is that Apple choose to symlink gcc and g++ to clang. Consequently, people kept using gcc/g++ in their makefiles. But clang is not gcc, it behaves differently, so I can understand why we chose not to emulate this behavior.

Comment (gridaphobe) [inline pkgs/tools/misc/bandwidth/default.nix]: cc-wrapper does set `CC` (https://github.com/NixOS/nixpkgs/blob/67fe5d46d4cd1c5383462bc136cd3c4ccc8cefc2/pkgs/build-support/cc-wrapper/setup-hook.sh#L42-L43).

I think the real problem is that the Makefile isn't inheriting the value of `CC` from the environment, so we probably need to make `CC=$CC` a default flag.

File: pkgs/development/libraries/libev/default.nix [hunk 3]
```diff
@@ -7,6 +7,11 @@ stdenv.mkDerivation rec {
     url = "http://dist.schmorp.de/libev/${name}.tar.gz";
     sha256 = "1jyw7qbl0spxqa0dccj9x1jsw7cj7szff43cq4acmklnra4mzz48";
   };
+
+  # Version 4.19 is not valid C11 (which Clang default to)
+  # Check if this is still necessary on upgrade
+  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.cc.isClang "-std=c99";
```
Comment (gridaphobe) [inline pkgs/development/libraries/libev/default.nix]: This use of `optionalString` is going to cause a mass-rebuild on linux, sadly. `optionalString` behaves like

```
optionalString b s = if b then s else ""
```

and the empty string gets put into the derivation, whereas in

```
if b then s else null
```

the `null` is filtered out, leaving the linux derivation unchanged :(

Comment (spwhitt) [inline pkgs/development/libraries/libev/default.nix]: Thanks, fixed.

**Pull request merged**